ext/phar/phar_object.c

```diff
@@ -573,7 +573,16 @@ PHP_METHOD(Phar, webPhar)
 	}
 
 	/* retrieve requested file within phar */
-	if (!(SG(request_info).request_method && SG(request_info).request_uri && (!strcmp(SG(request_info).request_method, "GET") || !strcmp(SG(request_info).request_method, "POST")))) {
+	if (!(SG(request_info).request_method
+          && SG(request_info).request_uri
+          && (!strcmp(SG(request_info).request_method, "DELETE")
+           || !strcmp(SG(request_info).request_method, "GET")
+           || !strcmp(SG(request_info).request_method, "HEAD")
+           || !strcmp(SG(request_info).request_method, "POST")
+           || !strcmp(SG(request_info).request_method, "PUT")
```

Comment:
Any particular reason to change the condition order? GET and POST were probably the common case, so would profit from returning earlier from the condition anyway.

Thanks.